src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -267,29 +272,47 @@ protected Future<Map<String, List<Piece>>> updateOrderAndPoLinesStatus(Map<Strin
     // Once all PO Lines are retrieved from storage check if receipt status
     // requires update and persist in storage
     return getPoLines(poLineIdsForUpdatedPieces, requestContext).compose(poLines -> {
-        // Calculate expected status for each PO Line and update with new one if required
-        // Skip status update if PO line status is Ongoing
-        List<Future<String>> futures = new ArrayList<>();
-        for (PoLine poLine : poLines) {
-          if (poLine.getReceiptStatus() == CANCELLED || poLine.getReceiptStatus() == ONGOING) {
-            continue;
-          }
-          List<Piece> successfullyProcessedPieces = getSuccessfullyProcessedPieces(poLine.getId(), piecesGroupedByPoLine);
-          futures.add(calculatePoLineReceiptStatus(poLine, successfullyProcessedPieces, requestContext)
-            .compose(status -> purchaseOrderLineService.updatePoLineReceiptStatus(poLine, status, requestContext)));
+      // Calculate expected status for each PO Line and update with new one if required
+      // Skip status update if PO line status is Ongoing or Cancelled
+      List<Future<PoLine>> poLinesToUpdate = new ArrayList<>();
+      for (PoLine poLine : poLines) {
+        if (poLine.getReceiptStatus() == CANCELLED || poLine.getReceiptStatus() == ONGOING) {
+          logger.info("updateOrderAndPoLinesStatus:: No pieces processed - POL with {} has status CANCELLED or ONGOING", poLine.getId());
+          continue;
         }
 
-        return collectResultsOnSuccess(futures).map(updatedPoLines -> {
-          logger.debug("{} out of {} PO Line(s) updated with new status", updatedPoLines.size(), piecesGroupedByPoLine.size());
+        List<Piece> successfullyProcessedPieces = getSuccessfullyProcessedPieces(poLine.getId(), piecesGroupedByPoLine);
+        if (CollectionUtils.isEmpty(successfullyProcessedPieces)) {
+          logger.info("updateOrderAndPoLinesStatus:: No pieces processed - nothing to update for POL with {}", poLine.getId());
+          continue;
+        }
 
-          List<PoLine> successPoLines = StreamEx.of(poLines)
-            .filter(line -> updatedPoLines.contains(line.getId()))
-            .toList();
-          updateOrderStatus.accept(successPoLines);
-          return null;
-        });
-      })
-      .map(ok -> piecesGroupedByPoLine);
+        Future<PoLine> poLineToUpdateFuture = getPiecesByPoLine(poLine.getId(), requestContext)
```

Comment:
Necessary for Receipt Status to be correctly determined

Reply:
can we use piecesFromStorage here that we already have in holder?

Reply:
> can we use piecesFromStorage here that we already have in holder?

From my debugging sessions, these pieces are not every piece for a given POL, if we use this one we will get inconsistent Receipt Status as it requires all pieces to be included in the calculation